toolchain/parser/parse_node_kind.def

```diff
@@ -77,7 +77,7 @@ CARBON_PARSE_NODE_KIND(PostfixOperator)
 // Struct literals.
 CARBON_PARSE_NODE_KIND(StructLiteral)
 CARBON_PARSE_NODE_KIND(StructTypeLiteral)
-CARBON_PARSE_NODE_KIND(StructLiteralOrTypeLiteralStart)
+CARBON_PARSE_NODE_KIND(StructLiteralOrStructTypeLiteralStart)
```

Comment:
I'm good with this, but would also be good with `StructOrStructTypeLiteralStart` which was the original instinct I had. Both sem less confusing than the other name for me. Happy for you to use whichever works best for you.

Reply:
(totally not important, all of this is fine)

FWIW, I was reading my suggestion as "(Struct or (Struct Type)) Literal Start" so that it still worked. ;] But yeah, the lack of precise grouping in an "or" name is always a bit rough.

I wonder if we should actually poke the design to give a common name to both of these literal syntaxes so that we can easily use that in places like this where without it we're forced to form awkward conjunctions. Anyways, that's for the future.